### PR TITLE
feat(search): migrate search endpoints functions to coroutines closes #224

### LINKED ISSUE #224: Migrate web handlers to Kotlin coroutines

### PR DESCRIPTION
**Pull request by philippestoltz**

### CODE REVIEW

**Review by bobeal — APPROVED** [review]

File: search-service/src/main/kotlin/com/egm/stellio/search/web/TemporalEntityHandler.kt [hunk 1]
```diff
@@ -103,34 +93,27 @@ class TemporalEntityHandler(
...
-                JsonLdUtils.filterCompactedEntityOnAttributes(it.compact(), temporalQuery.attrs)
-            }
-            .map {
-                ResponseEntity.status(HttpStatus.OK).body(serializeObject(it))
-            }
+        val temporalEntityAttribute = temporalEntityAttributeService.getForEntity(
+            entityId.toUri(),
+            temporalQuery.attrs,
+            contextLink
+        ).awaitFirstOrNull() ?: throw ResourceNotFoundException("Entity $entityId was not found")
```
Comment (HoucemKacem) [inline search-service/src/main/kotlin/com/egm/stellio/search/web/TemporalEntityHandler.kt:102]: I think you can use the ENTITY_NOT_FOUND_MESSAGE in ApiResponses.kt in shared module.

Comment (philippestoltz) [reply search-service/src/main/kotlin/com/egm/stellio/search/web/TemporalEntityHandler.kt:102]: OK , I am adding a function in ApiResponses.kt for the message including entityId

Comment (HoucemKacem) [reply search-service/src/main/kotlin/com/egm/stellio/search/web/TemporalEntityHandler.kt]: you can use it in TemporalEntityHandlerTests too 
`it should return a 404 if temporal entity attribute does not exist`

File: shared/src/main/kotlin/com/egm/stellio/shared/util/ApiResponses.kt [hunk 2]
```diff
@@ -1,3 +1,5 @@
 package com.egm.stellio.shared.util
 
 const val ENTITY_NOT_FOUND_MESSAGE = "Entity Not Found"
+
+fun entityNotFoundMessage(entityId: String) = "Entity $entityId was not found"
```
Comment (bobeal) [inline shared/src/main/kotlin/com/egm/stellio/shared/util/ApiResponses.kt:5]: don't be selfish, migrate the existing uses of `ENTITY_NOT_FOUND_MESSAGE` (and avoid a duplicated message)

Comment (philippestoltz) [reply shared/src/main/kotlin/com/egm/stellio/shared/util/ApiResponses.kt:5]: OK ;-)

File: search-service/src/main/kotlin/com/egm/stellio/search/web/TemporalEntityHandler.kt (hunk 1 above)
Comment (philippestoltz) [reply search-service/src/main/kotlin/com/egm/stellio/search/web/TemporalEntityHandler.kt]: OK

**Review by bobeal — APPROVED** [review]

**Review by HoucemKacem — APPROVED** [review]

**Pull request merged**